### PR TITLE
Added FXIOS-7460 [v120] enable XCode 15

### PR DESCRIPTION
**Pull request by OrlaM**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7460)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
Fixed some warnings when running on XCode 15.
This update has some breaking changes for XCode 14, so it looks like we will have to coordinate timing the update after all. cc @clarmso 
There were some linker warnings that I was unable to resolve, lot of other people online reporting similar so it seems like an XCode bug, to resolve this I have reverted to using the classic linker.

Keeping this as draft for now as we'll need to figure out the best time to merge. Also I could use some help figuring out if there is a way to test building this with XCode 15 on Bitrise without disturbing the existing workflows. cc @isabelrios 

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by OrlaM** [comment]
Awesome, ty!

**Comment by OrlaM** [comment]
Any idea what might cause this error? 
image

**Comment by clarmso** [comment]
The following unit tests fail on my end. Is the failure related to the ugprade?
```
TabTrayViewControllerTests.testToolbarItems_ForCompact()
TabTrayViewControllerTests.testToolbarItems_ForRegular()
RustPlacesTests.testHistoryMetadataBasics()
```

**Comment by clarmso** [comment]
> Any idea what might cause this error? image

The build fails on bitrise, but I could build on my computer successfully. Looking 👀 
https://app.bitrise.io/build/9373fae2-b3a0-4e98-ba57-5aa28b70e690?tab=log

**Comment by isabelrios** [comment]
> > Any idea what might cause this error? image
> 
> The build fails on bitrise, but I could build on my computer successfully. Looking 👀 https://app.bitrise.io/build/9373fae2-b3a0-4e98-ba57-5aa28b70e690?tab=log

The testing step failed and so there could not be a test report created, that's the error shown in the test report in Bitrise. Not very nice, we can share this with them. I guess it was a temporary issue because I see the tests running in next runs.

However, as @clarmso mentions, there are two unit tests failures in the last run I see on Bitrise:

ClientTests / TabTrayViewControllerTests: 

- testToolbarItems_ForRegular()
- testToolbarItems_ForCompact()

Looks like these tests fail consistently so they may need an update to work with the new Xcode15

**Comment by isabelrios** [comment]
Also, there are 10 warnings now (the threshold is 8), the build will fail if we don't address them.

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:182]: I'm not sure about this change... for the Unit test it is Fennec_UnitTest_iphonesimulator17.0-arm64.xctestrun... it has always been the same for all except fo the test plan name part...

**Comment by isabelrios** [comment]
This is almost ready! there is only one UI Test that may need to be updated to work on xcode 15.
- testSharePageWithShareSheetOptions()

Looking to update it

**Comment by isabelrios** [comment]
I have updated the UI Test that was failing, and with that, this PR would be ready to land.
@clarmso we would need to work on the FullFunctional as same as that test, there might be others that would need a small update.

**Comment by isabelrios** [comment]
FYi: looks like there might be performance regressions with Xcode 15: https://discuss.bitrise.io/t/xcode-15-performance-regressions/23385

**Comment by isabelrios** [comment]
Test failure is intermittent, test passed in a re-try, bitrise is green for the latest commit on this PR [53973f9](https://github.com/mozilla-mobile/firefox-ios/pull/16626/commits/53973f9891651bee4e3bab8410f2197513c8f34d)

**Comment by OrlaM** [comment]
Everything is passing on BR so I think this is ready to merge. Once I get some approvals on the PR I'll wait until after hours to hit merge to minimize disruption.

**Comment by clarmso** [comment]
> LGTM, When it lands we will be monitoring the build + tests. As I commented in this PR looks like there are performance regression issues with Xcode 15. We will upgrade to the next dot release as soon as it is available

Yes, I could live with the performance issue (21 min instead of 17 min spent on Bitrise pipeline).

**Review by yoanarios — APPROVED** [review]
LGTM

**Comment by clarmso** [comment]
😄 Let's merge this PR first. I can address the following test failures from the full functional test suite.
```
Failing tests:
        BookmarksTests.testAddNewFolder()
        DownloadsTests.testLongPressOnDownloadedFile()
        DownloadsTests.testRemoveUserDataRemovesDownloadedFiles()
        FindInPageTests.testFindFromSelection()
        SearchSettingsUITests.testCustomSearchEngineIsEditable()
        SearchSettingsUITests.testDeletingLastCustomEngineExitsEditing()
        ThirdPartySearchTest.testCustomSearchEngineDeletion()
        ThirdPartySearchTest.testCustomSearchEngines()
        ToolbarTests.testRevealToolbarWhenTappingOnStatusbar()
```

**Review by clarmso — APPROVED** [review]

**Comment by isabelrios** [comment]
@OrlaM oh I was about to comment if we could hold off on merging this PR...
- we need to know what's happening here: https://mozilla.slack.com/archives/C05C9RET70F/p1697014935731159

we may need to revert this change until we debug that issue to not to introduce more variables :/

**Pull request merged**
